atlasdb-config/src/main/java/com/palantir/atlasdb/config/BatcherProvider.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.config;
+
+import com.palantir.lock.client.InternalMultiClientConjureTimelockService;
+import java.util.function.Supplier;
+
+public interface BatcherProvider<T> {
+    T getBatcher(Supplier<InternalMultiClientConjureTimelockService> multiClientConjureTimelockService);
```

Comment:
Feels a bit weird - very specific arg but generic return type. I wonder what thoughts are on making this:
```java
T getBatcher(Supplier<K> service);
```
I can see pros and cons both ways. However, if you want to keep this param as is, maybe I'd be a bit more specific on the name (MultiClientBatchProvider? TimelockBatchProvider? idk).